src/coreclr/tools/Common/Internal/Text/Utf8StringBuilder.cs

```diff
@@ -16,10 +16,10 @@ public Utf8StringBuilder()
         {
         }
 
-        // TODO: This should return ReadOnlySpan<byte> instead once available
-        public byte[] UnderlyingArray => _buffer;
         public int Length => _length;
 
+        public ReadOnlySpan<byte> AsSpan() => _buffer;
```

Comment:
```suggestion
        public ReadOnlySpan<byte> AsSpan() => `_buffer.AsSpan(0, _length);
```
I would expect AsSpan() to only return the filled-in portion of the buffer. The existing use of `UnderlyingArray` in DwarfEhFrame.cs seems to have a bug.

Reply:
Thanks for keen eyed review!

Reply:
Now there's probably some length calculation failing due to fixing of the bug. Investigating..

Reply:
https://github.com/dotnet/runtime/pull/102101/files#diff-3e89bfe7a333612995a1cdbc5985119530130b8a7c639012dce8bd239dcaa583L84 may need to write an extra zero byte.

cc @filipnavara

Reply:
for reference (from https://www.corsix.org/content/elf-eh-frame)

```
struct cie {
  uint32_t length;           // In bytes, of all subsequent fields
  int32_t  zero;             // Must be zero
  uint8_t  version;          // Typically 1 or 3, sometimes 4
  char     aug_string[];     // NUL terminated
  if (aug_string[0] == 'e' && aug_string[1] == 'h') {
    void* eh_ptr;            // Only used by very old g++
  }
  if (version >= 4) {
    uint8_t addr_size;       // Must be sizeof(void*)
    uint8_t segment_size;    // Must be zero
  }
  uleb128 code_align;        // Typically 1 (even on aarch64)
  sleb128 data_align;        // Typically -4 or -8
  if (version == 1) {
    uint8_t return_address_ordinal;
  } else {
    uleb128 return_address_ordinal;
  }
  uint8_t aug_operands[];    // Relates to aug_string
  uint8_t dw_cfa_bytecode[];
  uint8_t zero_padding[];    // To give 4 or 8 byte alignment
};
```

Reply:
Possibly missing null-terminator now?